### ISSUE TITLE
"Cannot read Proerty '0' of undefined" After upgrade to 2.0.0

### ISSUE THREAD

**Issue by TomLingham** [power-assert-js/espower#33]
Hi, seem to be having issues after upgrading to 2.0.0. Our only dependencies are the intelli-espower-loader and power-assert, however this seems to have an issue that's only popped up in the last day or so. Possibly since the 2.0.0 release.

intelli-espower-loader: 1.0.1
power-assert: 1.4.2

```
app/node_modules/espower/lib/assertion-visitor.js:372
    var firstBody = body[0];
                        ^
TypeError: Cannot read property '0' of undefined
    at insertAfterUseStrictDirective (/home/s98430/Code/nova/nova-service-diagnostic/app/node_modules/espower/lib/assertion-visitor.js:372:25)
    at /home/s98430/Code/nova/nova-service-diagnostic/app/node_modules/espower/lib/assertion-visitor.js:222:9
    at /home/s98430/Code/nova/nova-service-diagnostic/app/node_modules/espower/lib/transformation.js:17:9
    at Array.forEach (native)
```

Thanks :smile:

**Comment by twada** [comment]
@TomLingham Thank you for reporting!
I'll take a look ASAP.

Would you create a small reproduction case for this issue?
I want to know your code and environment.

**Comment by TomLingham** [comment]
Alright, so after further digging around I also discovered this error only happens if the `directories.test` field is set to an existing folder in package.json. Here is a sample project that you should be able to just pull down and run `npm test` on. If you remove the `test` key in the package.json the error no longer happens.
https://github.com/TomLingham/espower-sample

I don't even know if this is a problem with your package but the error shows up in espower :confused:

**Comment by TomLingham** [comment]
So, apparently that needs to be there for instrumentation... Hmm...

**Comment by TomLingham** [comment]
Alright, inline functions seem to be causing the problem, assuming that the instrumentation is looking for the body of a function (but there is none) which is why it is having the problem.

If you run `npm run test` and npm run test-fixed` on this repo you should see the differences.

**Comment by twada** [comment]
@TomLingham Ah, inline functions!
Your repro case helps me a lot. Thanks!!

I have to deal with `ArrowFunctionExpression`s that have `CallExpression` as its immediate child.

**Comment by twada** [comment]
Just released [espower 2.0.1](https://github.com/power-assert-js/espower/blob/master/CHANGELOG.md#201-2016-11-16) to fix this issue.
Thank you for your reporting!

**Comment by TomLingham** [comment]
Thank you! It works great! :)

On Wed., 16 Nov. 2016, 7:05 pm Takuto Wada, <notifications@github.com>
wrote:

Just released espower 2.0.1
<https://github.com/power-assert-js/espower/blob/master/CHANGELOG.md#201-2016-11-16>
to fix this issue.
Thank you for your reporting!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/power-assert-js/espower/issues/33#issuecomment-260879897>,
or mute the thread

**Comment by twada** [comment]
@TomLingham Glad to hear that 😃
